pkg/theia/commands/anomaly_detection_algos.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	anomalydetector "antrea.io/theia/pkg/apis/anomalydetector/v1alpha1"
+	"antrea.io/theia/pkg/theia/commands/config"
+	"antrea.io/theia/pkg/util"
+)
+
+// throughputAnomalyDetectionEWMACmd represents the policy-recommendation delete command
+var throughputAnomalyDetectionAlgoCmd = &cobra.Command{
+	Use:   "run",
+	Short: "throughput Anomaly detection using Algo",
+	Long:  `throughput Anomaly detection using Algorithms, currently supported Algorithms are EWMA, ARIMA and DBSCAN`,
+	Example: `Run the spefic Algorithm for throughput Anomaly Detection
+	$ theia throughput-anomaly-detection run --algo ARIMA --start-time 2022-01-01T00:00:00 --end-time 2022-01-31T23:59:59
+	Run thoughput anomaly detection algorithm of type ARIMA starting from '2022-01-01 00:00:00' until '2022-01-31 23:59:59'`,
+	RunE: throughputAnomalyDetectionAlgo,
+}
+
+var (
+	algoName string
+)
+
+func throughputAnomalyDetectionAlgo(cmd *cobra.Command, args []string) error {
+	throughputAnomalyDetection := anomalydetector.ThroughputAnomalyDetector{}
+	algoName, err := cmd.Flags().GetString("algo")
+	if err != nil {
+		return err
+	}
+	if algoName == "" && len(args) == 1 {
+		algoName = args[0]
+	}
```

Comment:
Are you trying to support command in this format `theia throughput-anomaly-detection run ARIMA`. But you have a check of `err := throughputAnomalyDetectionAlgoCmd.MarkFlagRequired("algo")` in `init`, then I guess this code won't work? And if you would like to support this, maybe also add the description in help message and doc

Reply:
ack